### PR TITLE
added keras tuner and updated readme

### PR DESCRIPTION
**Pull request by Kanavpreet-Singh**
worked on issue #41

### CODE REVIEW

**Comment by Cgarg9** [comment]
@Kanavpreet-Singh well done so far. Small updates - 

1. Lets make a directory for your code inside the model tuning folder.. it should have two files : one for your notebook and one for the assignment
2. Some code blacks have not been run. please run it completely so that everyone can easily visualise what you aimed to do at each step.

**Comment by Kanavpreet-Singh** [comment]
@Cgarg9 ive tried to implement the suggestions , kindly check and update me .

**Comment by Cgarg9** [comment]
Screenshot 2025-04-04 at 11 14 48 AM
from where are we getting these statements?
also kindly delete the test.ipynb

**Comment by Kanavpreet-Singh** [comment]
@Cgarg9 , When you call tuner.search(x_train, y_train, epochs=3, validation_split=0.2, callbacks=[early_stopping]), Keras Tuner starts running different trials with different hyperparameters. It automatically logs the results of each trial, including validation accuracy, best accuracy so far, and elapsed time. ive removed test.ipynb file

**Comment by Cgarg9** [comment]
Got it but why is your print statement not present?

**Comment by Kanavpreet-Singh** [comment]
@Cgarg9 sir ive again tried running the cells, 
initially it does show the print statement like in the screenshot below
Screenshot 2025-04-05 074630

but afterwards , when trials run , then it only shows info of the particular trial like shown in screenshot below
Screenshot 2025-04-05 075510

At the end , trial 5 ran , so it ended up showing its info

**Comment by Cgarg9** [comment]
Okay got it. I'll look into why this is happening later but we can merge this. Great job.
